quantum/rgblight.h

```diff
@@ -182,7 +182,20 @@ typedef struct {
 #        define RGBLIGHT_END_SEGMENT_INDEX (255)
 #        define RGBLIGHT_END_SEGMENTS \
             { RGBLIGHT_END_SEGMENT_INDEX, 0, 0, 0 }
-#        define RGBLIGHT_MAX_LAYERS 8
+#        ifndef RGBLIGHT_MAX_LAYERS
+#            define RGBLIGHT_MAX_LAYERS 8
+#        endif
+#        if RGBLIGHT_MAX_LAYERS <= 0
+#error invalid RGBLIGHT_MAX_LAYERS value (must be >= 1)
+#        elif RGBLIGHT_MAX_LAYERS <= 8
+typedef uint8_t rgblight_layer_mask_t;
+#        elif RGBLIGHT_MAX_LAYERS <= 16
+typedef uint16_t rgblight_layer_mask_t;
+#        elif RGBLIGHT_MAX_LAYERS <= 32
+typedef uint32_t rgblight_layer_mask_t;
+#        else
+#error invalid RGBLIGHT_MAX_LAYERS value (must be <= 32)
+#        endif
```

Comment:
To continue from my previous comment... 

It may be a good idea to check `LAYER_STATE_#BIT` status, and limit this based on that. 

That way, this can never exceed the max layers allowed.  

I know that adds additional complexity here, but may prevent issues down the line. 

```suggestion
#        ifdef LAYER_STATE_8BIT && RGBLIGHT_MAX_LAYERS > 8
#            undef RGBLIGHT_MAX_LAYERS
#            define RGBLIGHT_MAX_LAYERS 8
#        elif LAYER_STATE_16BIT && RGBLIGHT_MAX_LAYERS > 16
#            undef RGBLIGHT_MAX_LAYERS
#            define RGBLIGHT_MAX_LAYERS 16
#        endif
#        ifndef RGBLIGHT_MAX_LAYERS
#            define RGBLIGHT_MAX_LAYERS 8
#        endif
#        if RGBLIGHT_MAX_LAYERS <= 0 || RGBLIGHT_MAX_LAYERS > 32
#error invalid RGBLIGHT_MAX_LAYERS value (must be between 0 and 32)
#        elif RGBLIGHT_MAX_LAYERS <= 8
typedef uint8_t rgblight_layer_mask_t;
#        elif RGBLIGHT_MAX_LAYERS <= 16
typedef uint16_t rgblight_layer_mask_t;
#        elif RGBLIGHT_MAX_LAYERS <= 32
typedef uint32_t rgblight_layer_mask_t;
#        endif
```

Reply:
I can think of several use cases where the number of lighting layers must exceed the number of keymap layers. For instance, suppose a user wants to use LL's to indicate keymap layer and to indicate capslock / numlock and perhaps for some other indicators. In own case, I have just 4 layers in my keymap, but at present count I have 10 lighting layers.

Reply:
I consider the keymap layer and the rgblight layer to be unrelated to each other. For example, you might use the rgblight layer only for displays such as CAPS_LOCK.

Reply:
Okay, sounds good.